fix: Catch correct ConnectionError for additional_chat_templates

akug opened the pull request:
# What does this PR do?

Catch the correct `ConnectionError` when checking for `additional_chat_templates`.

Fixes #39873 


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

akug committed:
fix: Catch correct ConnectionError for additional_chat_templates

Rocketknight1 commented:
Good PR, but I'm not sure about timeouts! This might cause strange issues when the user has an internet connection.

However, you're totally right that we used the native Python `ConnectionError` instead of the `ConnectionError` in `requests`. What's the reason for including timeouts in the `except` clause?

akug committed:
fix: don't catch timeout

akug committed:
fix: formatting

akug commented:
> Good PR, but I'm not sure about timeouts! This might cause strange issues when the user has an internet connection.
> 
> However, you're totally right that we used the native Python `ConnectionError` instead of the `ConnectionError` in `requests`. What's the reason for including timeouts in the `except` clause?

Thanks for the quick feedback and sorry for the late reply! I added the TimeoutError because I have seen it in the `has_file` function in the same file, and assumed it might be necessary here as well. But after your comment I thought more about it and I think you are right. 

I removed catching the TimeoutError now, please have a look again when you find the time :)

Rocketknight1 reviewed (approved):
This should be ready to merge now! I'm just fighting the CI to get it to be green, lol
